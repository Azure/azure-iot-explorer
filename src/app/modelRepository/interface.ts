import { ModelRepositoryStateInterface } from "../shared/modelRepository/state";
import { StringMap } from '../api/models/stringMap';

export interface ModelRepositoryFormOpsInterface {
    setDirtyFlag: (dirty: boolean) => void;
    setRepositoryLocationSettings: (settings: ModelRepositoryStateInterface) => void;
    setRepositoryLocationSettingsErrors: (errors: StringMap<string> ) => void;
}

export const getInitialModelRepositoryFormOps = (): ModelRepositoryFormOpsInterface => ({
    setDirtyFlag: () => undefined,
    setRepositoryLocationSettings:  () => undefined,
    setRepositoryLocationSettingsErrors:  () => undefined
});