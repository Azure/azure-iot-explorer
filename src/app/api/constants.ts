/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
export enum HTTP_OPERATION_TYPES {
    Delete = 'DELETE',
    Get = 'GET',
    Patch = 'PATCH',
    Post = 'POST',
    Put = 'PUT'
}

export const APPLICATION_JSON = 'application/json';
export const ERROR_TYPES = {
    HTTP: 'http'
};
