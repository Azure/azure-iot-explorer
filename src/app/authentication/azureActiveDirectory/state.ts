/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import { IotHubDescription } from '../../api/models/iotHubDescription';
import { AzureSubscription } from '../../api/models/azureSubscription';
export interface AzureActiveDirectoryStateInterface {
    formState: 'initialized' | 'working' | 'failed' | 'idle' | 'keyPicked';
    iotHubKey: string;
    iotHubs: IotHubDescription[];
    subscriptions: AzureSubscription[];
    token: string;
}

export const getInitialAzureActiveDirectoryState = (): AzureActiveDirectoryStateInterface => ({
    formState: 'initialized',
    iotHubKey: undefined,
    iotHubs: [],
    subscriptions: [],
    token: undefined
});
