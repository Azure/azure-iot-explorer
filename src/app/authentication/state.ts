/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
export enum AuthenticationMethodPreference {
    AzureAD = 'AzureAD',
    ConnectionString = 'ConnectionString'
}

export interface AuthenticationStateInterface {
    formState: 'initialized' | 'working' | 'failed' | 'idle';
    preference: string;
}

export const getInitialAuthenticationState = (): AuthenticationStateInterface => ({
    formState: 'initialized',
    preference: undefined
});
