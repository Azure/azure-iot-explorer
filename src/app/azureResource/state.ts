/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import { AzureResource } from './models/azureResource';

export interface AzureResourceStateInterface {
    activeAzureResource?: AzureResource;
}

export const azureResourceStateInitial = (): AzureResourceStateInterface => {
    return {};
};
