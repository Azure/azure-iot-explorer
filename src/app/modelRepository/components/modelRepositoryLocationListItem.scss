/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
 @import '../../css/themes';

.folder-picker-dialog {
    .ms-Dialog-main {
        min-width: 50vw;
        max-width: 356px;
        min-height: 30vh;
    }
    .folder-links{
        height: 30vh;
        overflow: auto;
    }
    .ms-Dialog-header{
        word-wrap: break-word;
    }
    .folder-button {
        width: 100%;
        text-align: left;
        border: none;
        margin-bottom: 10px;
    }
    .no-folders-text {
        margin-left: 22px;
    }
}

.local-folder-launch {
    margin-top: 5px;
    float: right;
}

.local-folder-textbox {
    .ms-Label {
        float:left;
        margin-bottom: 10px;
        margin-right: 5px;
    }
    .ms-TextField-field {
        @include themify($themes) {
            background-color: themed('maskedCopyableReadonlyBackground');
        }
    }
}
