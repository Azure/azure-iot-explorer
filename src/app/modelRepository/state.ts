import { ModelRepositoryStateInterface } from "../shared/modelRepository/state";
import { StringMap } from '../api/models/stringMap';
import { REPOSITORY_LOCATION_TYPE } from "../constants/repositoryLocationTypes";

export interface ModelRepositoryFormStateInterface {
    dirty: boolean;
    repositoryLocationSettings: ModelRepositoryStateInterface;
    repositoryLocationSettingsErrors: StringMap<string>;
}

export const getInitialModelRepositoryFormState = (): ModelRepositoryFormStateInterface => ({
    dirty: false,
    repositoryLocationSettings: [{repositoryLocationType: REPOSITORY_LOCATION_TYPE.Public, value: '' }],
    repositoryLocationSettingsErrors: {}
});