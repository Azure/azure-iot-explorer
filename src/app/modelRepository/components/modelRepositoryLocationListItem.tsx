/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { IconButton } from '@fluentui/react';
import { REPOSITORY_LOCATION_TYPE } from '../../constants/repositoryLocationTypes';
import { ResourceKeys } from '../../../localization/resourceKeys';
import { CANCEL } from '../../constants/iconNames';
import { ModelRepositoryConfiguration } from '../../shared/modelRepository/state';
import { ModelRepositoryFormType } from '../hooks/useModelRepositoryForm';
import { ListItemPublicRepo } from './listItemPublicRepo';
import { ListItemConfigurableRepo } from './listItemConfigurableRepo';
import { ListItemLocal } from './listItemLocal';
import './modelRepositoryLocationListItem.scss';

export interface ModelRepositoryLocationListItemProps {
    index: number;
    item: ModelRepositoryConfiguration;
    formState: ModelRepositoryFormType;
}

export const ModelRepositoryLocationListItem: React.FC<ModelRepositoryLocationListItemProps> = ({index, item, formState}) => {
    const { t } = useTranslation();
    const [{repositoryLocationSettings }, {setRepositoryLocationSettings, setDirtyFlag}] = formState;

    const onRemove = () => {
        const updatedRepositoryLocationSettings = [...repositoryLocationSettings];
        updatedRepositoryLocationSettings.splice(index, 1);

        setDirtyFlag(true);
        setRepositoryLocationSettings(updatedRepositoryLocationSettings);
    };

    const renderItemDetail = () => {
        return (
            <div className="item-details">
                {item.repositoryLocationType === REPOSITORY_LOCATION_TYPE.Public &&
                    <ListItemPublicRepo/>
                }
                {(item.repositoryLocationType === REPOSITORY_LOCATION_TYPE.Local || item.repositoryLocationType === REPOSITORY_LOCATION_TYPE.LocalDMR) &&
                    <ListItemLocal
                        index={index}
                        item={item}
                        formState={formState}
                        repoType={item.repositoryLocationType}
                    />
                }
                {item.repositoryLocationType === REPOSITORY_LOCATION_TYPE.Configurable &&
                    <ListItemConfigurableRepo
                        index={index}
                        item={item}
                        formState={formState}
                    />
                }
            </div>);
    };

    return (
        <div className="item" role="list">
            <div className="numbering">
                {index + 1}
            </div>
            <div
                className="location-item"
                role="listitem"
            >
                {renderItemDetail()}
                <IconButton
                    className="remove-button"
                    iconProps={{ iconName: CANCEL }}
                    title={t(ResourceKeys.modelRepository.commands.remove.label)}
                    ariaLabel={t(ResourceKeys.modelRepository.commands.remove.ariaLabel)}
                    onClick={onRemove}
                />
            </div>
        </div>
    );
};
