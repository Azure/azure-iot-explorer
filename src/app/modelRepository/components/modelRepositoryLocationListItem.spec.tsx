/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import 'jest';
import * as React from 'react';
import { shallow } from 'enzyme';
import { IconButton } from '@fluentui/react';
import { act } from 'react-dom/test-utils';
import { ModelRepositoryLocationListItem } from './modelRepositoryLocationListItem';
import { REPOSITORY_LOCATION_TYPE } from '../../constants/repositoryLocationTypes';
import { getInitialModelRepositoryFormState } from '../state';
import { getInitialModelRepositoryFormOps } from '../interface';

describe('ModelRepositoryLocationListItem', () => {
    it('matches snapshot for public repo', () => {
        const wrapper = shallow(
            <ModelRepositoryLocationListItem
                index={0}
                item={{
                    repositoryLocationType: REPOSITORY_LOCATION_TYPE.Public,
                    value: ''
                }}
                formState={[getInitialModelRepositoryFormState(), getInitialModelRepositoryFormOps()]}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('matches snapshot for configurable repo', () => {
        const wrapper = shallow(
            <ModelRepositoryLocationListItem
                index={1}
                item={{
                    repositoryLocationType: REPOSITORY_LOCATION_TYPE.Configurable,
                    value: 'test.com'
                }}
                formState={[getInitialModelRepositoryFormState(), getInitialModelRepositoryFormOps()]}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('matches snapshot for local folder', () => {
        const wrapper = shallow(
            <ModelRepositoryLocationListItem
                index={2}
                item={{
                    repositoryLocationType: REPOSITORY_LOCATION_TYPE.Local,
                    value: 'c:/models'
                }}
                formState={[getInitialModelRepositoryFormState(), getInitialModelRepositoryFormOps()]}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('matches snapshot for local repo', () => {
        const wrapper = shallow(
            <ModelRepositoryLocationListItem
                index={3}
                item={{
                    repositoryLocationType: REPOSITORY_LOCATION_TYPE.LocalDMR,
                    value: 'c:/dtmi'
                }}
                formState={[getInitialModelRepositoryFormState(), getInitialModelRepositoryFormOps()]}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('remove item calls expected operations', () => {
        const setRepositoryLocationSettings = jest.fn();
        const setDirtyFlag = jest.fn();
        const wrapper = shallow(
            <ModelRepositoryLocationListItem
                index={0}
                item={{
                    repositoryLocationType: REPOSITORY_LOCATION_TYPE.Public,
                    value: ''
                }}
                formState={[getInitialModelRepositoryFormState(), {...getInitialModelRepositoryFormOps(), setRepositoryLocationSettings, setDirtyFlag}]}
            />
        );
        const removeButton = wrapper.find(IconButton).first();
        act(() => removeButton.props().onClick(undefined));

        expect(setRepositoryLocationSettings).toBeCalledWith([]);
        expect(setDirtyFlag).toBeCalledWith(true);
    });
});
