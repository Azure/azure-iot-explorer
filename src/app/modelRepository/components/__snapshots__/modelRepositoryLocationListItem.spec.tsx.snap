// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelRepositoryLocationListItem matches snapshot for configurable repo 1`] = `
<div
  className="item"
  role="list"
>
  <div
    className="numbering"
  >
    2
  </div>
  <div
    className="location-item"
    role="listitem"
  >
    <div
      className="item-details"
    >
      <ListItemConfigurableRepo
        formState={
          Array [
            Object {
              "dirty": false,
              "repositoryLocationSettings": Array [
                Object {
                  "repositoryLocationType": "PUBLIC",
                  "value": "",
                },
              ],
              "repositoryLocationSettingsErrors": Object {},
            },
            Object {
              "setDirtyFlag": [Function],
              "setRepositoryLocationSettings": [Function],
              "setRepositoryLocationSettingsErrors": [Function],
            },
          ]
        }
        index={1}
        item={
          Object {
            "repositoryLocationType": "CONFIGURABLE",
            "value": "test.com",
          }
        }
      />
    </div>
    <CustomizedIconButton
      ariaLabel="modelRepository.commands.remove.ariaLabel"
      className="remove-button"
      iconProps={
        Object {
          "iconName": "Cancel",
        }
      }
      onClick={[Function]}
      title="modelRepository.commands.remove.label"
    />
  </div>
</div>
`;

exports[`ModelRepositoryLocationListItem matches snapshot for local folder 1`] = `
<div
  className="item"
  role="list"
>
  <div
    className="numbering"
  >
    3
  </div>
  <div
    className="location-item"
    role="listitem"
  >
    <div
      className="item-details"
    >
      <ListItemLocal
        formState={
          Array [
            Object {
              "dirty": false,
              "repositoryLocationSettings": Array [
                Object {
                  "repositoryLocationType": "PUBLIC",
                  "value": "",
                },
              ],
              "repositoryLocationSettingsErrors": Object {},
            },
            Object {
              "setDirtyFlag": [Function],
              "setRepositoryLocationSettings": [Function],
              "setRepositoryLocationSettingsErrors": [Function],
            },
          ]
        }
        index={2}
        item={
          Object {
            "repositoryLocationType": "LOCAL",
            "value": "c:/models",
          }
        }
        repoType="LOCAL"
      />
    </div>
    <CustomizedIconButton
      ariaLabel="modelRepository.commands.remove.ariaLabel"
      className="remove-button"
      iconProps={
        Object {
          "iconName": "Cancel",
        }
      }
      onClick={[Function]}
      title="modelRepository.commands.remove.label"
    />
  </div>
</div>
`;

exports[`ModelRepositoryLocationListItem matches snapshot for local repo 1`] = `
<div
  className="item"
  role="list"
>
  <div
    className="numbering"
  >
    4
  </div>
  <div
    className="location-item"
    role="listitem"
  >
    <div
      className="item-details"
    >
      <ListItemLocal
        formState={
          Array [
            Object {
              "dirty": false,
              "repositoryLocationSettings": Array [
                Object {
                  "repositoryLocationType": "PUBLIC",
                  "value": "",
                },
              ],
              "repositoryLocationSettingsErrors": Object {},
            },
            Object {
              "setDirtyFlag": [Function],
              "setRepositoryLocationSettings": [Function],
              "setRepositoryLocationSettingsErrors": [Function],
            },
          ]
        }
        index={3}
        item={
          Object {
            "repositoryLocationType": "DMR",
            "value": "c:/dtmi",
          }
        }
        repoType="DMR"
      />
    </div>
    <CustomizedIconButton
      ariaLabel="modelRepository.commands.remove.ariaLabel"
      className="remove-button"
      iconProps={
        Object {
          "iconName": "Cancel",
        }
      }
      onClick={[Function]}
      title="modelRepository.commands.remove.label"
    />
  </div>
</div>
`;

exports[`ModelRepositoryLocationListItem matches snapshot for public repo 1`] = `
<div
  className="item"
  role="list"
>
  <div
    className="numbering"
  >
    1
  </div>
  <div
    className="location-item"
    role="listitem"
  >
    <div
      className="item-details"
    >
      <ListItemPublicRepo />
    </div>
    <CustomizedIconButton
      ariaLabel="modelRepository.commands.remove.ariaLabel"
      className="remove-button"
      iconProps={
        Object {
          "iconName": "Cancel",
        }
      }
      onClick={[Function]}
      title="modelRepository.commands.remove.label"
    />
  </div>
</div>
`;
